### ISSUE TITLE
vpp crashes different with fix for #757 when run in hostNetwork:false

### ISSUE THREAD

**Issue by edwarnicke** [networkservicemesh/networkservicemesh#775]
I think the fix to #757 was likely valid, but we are now hitting an issue with a crash that manifests when running that fix in a pod with hostNetwork:false:

The traceback is:
```
/usr/bin/vpp[17]: buffer: vlib_physmem_shared_map_create: pmalloc_map_pages: Unable to fulfill huge page allocation request: No such file or directory

/usr/bin/vpp[17]: buffer: falling back to non-hugepage backed buffer pool
/usr/bin/vpp[17]: vlib_physmem_shared_map_create: pmalloc_map_pages: Unable to fulfill huge page allocation request: No such file or directory
/usr/bin/vpp[17]: received signal SIGSEGV, PC 0x7f9b5251dfee, faulting address 0xa0
/usr/bin/vpp[17]: #0  0x00007f9b51b1b97d 0x7f9b51b1b97d
/usr/bin/vpp[17]: #1  0x00007f9b51671890 0x7f9b51671890
/usr/bin/vpp[17]: #2  0x00007f9b5251dfee 0x7f9b5251dfee
/usr/bin/vpp[17]: #3  0x00007f9b51addaed vlib_call_init_exit_functions + 0x5d
/usr/bin/vpp[17]: #4  0x00007f9b51af06a9 vlib_main + 0xc9
/usr/bin/vpp[17]: #5  0x00007f9b51b1aa56 0x7f9b51b1aa56
/usr/bin/vpp[17]: #6  0x00007f9b513f1004 0x7f9b513f1004
```

The collateral is:
* [coredump](https://drive.google.com/file/d/17JISExefko8oYD7aDDjtkvVEdzXbemPS/view?usp=sharing)
* [deb files](https://drive.google.com/file/d/10o9yCXfwInFNFpPYX0chepujQML3jDtq/view?usp=sharing)

The deb files where constructed by applying 
[mincorefix.txt](https://github.com/networkservicemesh/networkservicemesh/files/2903381/mincorefix.txt)

to commit 6b4a32ded6d3db56d886e79154e8329bdfc0ed2a in the vpp repo.

The resulting docker file from building this with vppagent can be found in docker hub as edwarnicke/vpp-agent:latest

#776 should have a nice setup to reproduce, it uses the edwarnicke/vpp-agent controller, which now works nicely for running with hostNetwork:true for vppagent-dataplane, but leads to this crash in vppagent-firewall-nse.  The vppagent-firewall-nse is setup to simply run tail -f /dev/null, so you can kubectl exec in and run /usr/bin/vpp -c /etc/vpp/vpp.conf to reproduce the crash.

One note, if I add

```
securityContext:
            privileged: true
```

 to [k8s/conf/vppagent-firewall-nse.yaml](https://github.com/networkservicemesh/networkservicemesh/blob/a1af867282bde848574b7d2061e5f5625ba611ed/k8s/conf/vppagent-firewall-nse.yaml#L14) the crashing stops, but we need to be able to run unprivileged VPP in containers.

**Comment by artem-belov** [comment]
https://gerrit.fd.io/r/#/c/17842/

**Comment by artem-belov** [comment]
If buffer allocation return error, exit handlers trying to release resources before  they are created. vlib_worker_thread_barrier_sync trying to get worker thread by null pointer and catch SIGSEGV

**Comment by edwarnicke** [comment]
Ok... do we know why the buffer handler is returning an error?

**Comment by artem-belov** [comment]
Yes. 4 Kb pages wasn't initialize without
```
securityContext:
            privileged: true
```
Fixed in #757 today

**Comment by artem-belov** [comment]
Fix is uploaded to master branch by commit
https://github.com/FDio/vpp/commit/8343ee5665942353e57ee074da62b9d07c1c510b
